submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/experiment/codelocalizer/GitCodeLocalizer.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.server.submitter.k8s.experiment.codelocalizer;
+
+import org.apache.submarine.server.api.spec.ExperimentSpec;
+
+import io.kubernetes.client.models.V1PodSpec;
+
+public abstract class GitCodeLocalizer extends AbstractCodeLocalizer {
```

Comment:
How we handle Git pull credentials, should it part of ExperimentSpec, or part of Namespace secret. 

@manirajv06, @liuxunorg , @tangzhankun: any ideas here?

Reply:
Probably, one time activity based approach is preferable as it doesn't change for every experiment. It remains same for repo's.

Reply:
The credential seems should be created in the workbench and couple with the user account. And we may also think about the refresh of the secure token.  I'd prefer to not do the secure localizer for now.